### PR TITLE
Reset master to PosV4

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

**Comment by dangershony** [comment]
This branch is failing to sync at block 198 with this error
```
info: Stratis.Bitcoin.Connection.ConnectionManagerBehavior[0]
      Peer '[::ffff:165.22.90.248]:46660' connected (outbound), agent 'x1d 1.1.7265:3.0.6 (70012)', height 5053
warn: Stratis.Bitcoin.Features.Consensus.Rules.PosConsensusRuleEngine[0]
      Block '199-981be6947578725dc515a150c7f9e9dcc5ec1d59d0cbd2de56b40f8f33740e12-PartiallyValidated - PH' failed full validation with error: 'bad-txns-in-belowout : input value below output value'.
warn: Stratis.Bitcoin.Features.Consensus.Rules.PosConsensusRuleEngine[0]
      Marking '981be6947578725dc515a150c7f9e9dcc5ec1d59d0cbd2de56b40f8f33740e12' invalid.
info: Stratis.Bitcoin.Connection.ConnectionManagerBehavior[0]
      Peer '[::ffff:165.22.90.248]:46660' (outbound) offline, reason: 'The peer was banned, reason: input value below output value.'
i
```

**Comment by aubergemediale** [comment]
no idea what that is at block 198...

**Comment by aubergemediale** [comment]
`"198": {
      "hashBlock": "40a90e0103a0fddd7da6f464668919b193a0f22cdf8e839ee10734e00cc5ace7",
      "transactions": [
        {
          "valueAdded": 500000000,
          "txType": "Coinbase",
          "hashTx": "f262a828a238553eb5c088e82263ff84b2c2a13257759844f71297ffc22cfb5d",
          "received": {
            "f262a828a238553eb5c088e82263ff84b2c2a13257759844f71297ffc22cfb5d-0": {
              "address": "odx1q0693fqjqze4h7jy44vpmp8qtpk8v2rws0xa486",
              "hashTx": "f262a828a238553eb5c088e82263ff84b2c2a13257759844f71297ffc22cfb5d",
              "index": 0,
              "satoshis": 500000000
            }
          }
        }
      ],
      "time": 1573237066
    },`

**Comment by aubergemediale** [comment]
block 198 is a normal coinbase tx only...

**Comment by aubergemediale** [comment]
I deleted my chain and synced again...it did not pause at 198

File: src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs [hunk 1]
```diff
@@ -328,7 +328,8 @@ public bool CheckStakeKernelHash(PosRuleContext context, uint headerBits, uint25
             {
                 var serializer = new BitcoinStream(ms, true);
                 serializer.ReadWrite(prevStakeModifier);
-                serializer.ReadWrite(stakingCoins.Time);
+                if (this.network.Consensus.ConsensusFactory is PosConsensusFactory) // to stay compatible with legacy ODN
```
Comment (aubergemediale) [inline src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs]: is PosConensusFactory is always true on PoS networks, if there is nTime, or not....

Comment (aubergemediale) [reply src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs]: instead one can check weather the PosCosensusFactory creates PosTransaction or Transaction instances....

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs]: Yeah probably more correct

**Pull request merged**